tools/omnissm/functions/es-upload/main.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 Capital One Services, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-xray-sdk-go/xray"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/olivere/elastic"
+	"github.com/sha1sum/aws_signing_client"
+)
+
+var (
+	xRayTracingEnabled = os.Getenv("_X_AMZN_TRACE_ID")
+	esClient           = os.Getenv("OMNISSM_ELASTIC_SEARCH_HTTP")
+	indexName          = os.Getenv("OMNISSM_INDEX_NAME")
+	typeName           = os.Getenv("OMNISSM_TYPE_NAME")
+	mappingBucket      = os.Getenv("OMNISSM_MAPPING_BUCKET")
+	mappingKey         = os.Getenv("OMNISSM_MAPPING_KEY")
+	s3Svc              = s3.New(session.New())
+)
+
+func main() {
+	lambda.Start(func(ctx context.Context, event events.S3Event) {
+		if xRayTracingEnabled != "" {
+			xray.AWS(s3Svc.Client)
+		}
+		if esClient == "" || indexName == "" || typeName == "" {
+			log.Fatal("Missing required env variables OMNISSM_ELASTIC_SEARCH_HTTP, OMNISSM_INDEX_NAME, OMNISSM_TYPE_NAME")
+		}
+		client, err := newElasticClient(esClient)
+		if err != nil {
+			log.Fatal(err)
+		}
+		exists, err := client.IndexExists(indexName).Do(context.Background())
```

Comment:
doing a round trip network check for a one time initialization in a hot path, seems less then ideal, index setup should just be a makefile target not inline to the event indexing.

Reply:
@kapilt I've updated it to have index checking and creation to be done in a separate script.